packages/graphback-runtime-knex/tests/data/KnexDbDataProviderTest.ts

```diff
@@ -736,3 +732,379 @@ type Todo {
   expect(deletedTodo.id).toEqual(3);
   expect(deletedTodo.text).toEqual("updated todo");
 });
+
+describe('or', () => {
+  it('a && (b || c)', async () => {
+    const { providers: { Todo } } = await setup(`
+    
+
+    """
+    @model
+    """
+    type Todo {
+      id: ID
+      a: Int
+      b: Int
+      c: Int
+    }
+    `, {
+      seedData: {
+        todo: [
+          {
+            a: 1,
+            b: 5,
+            c: 8
+          },
+          {
+            a: 1,
+            b: 2,
+            c: 10
+          },
+          {
+            a: 1,
+            b: 5,
+            c: 3
+          },
+          {
+            a: 6,
+            b: 6,
+            c: 3
+          }
+        ]
+      }
+    })
+
+    const filter: QueryFilter = {
+      a: {
+        eq: 1
+      },
+      or: [
+        {
+          c: {
+            eq: 6
+          }
+        }, {
+          b: {
+            eq: 5
+          }
+        }
+      ]
+    }
+
+    const items = await Todo.findBy({ filter });
+
+    expect(items).toHaveLength(2);
+  });
+
+  it('a && (b || c) starting at first $or', async () => {
+    const { providers: { Todo } } = await setup(`
+    
+
+    """
+    @model
+    """
+    type Todo {
+      id: ID
+      a: Int
+      b: Int
+      c: Int
+    }
+    `, {
+      seedData: {
+        todo: [
+          {
+            a: 1,
+            b: 5,
+            c: 8
+          },
+          {
+            a: 1,
+            b: 2,
+            c: 10
+          },
+          {
+            a: 1,
+            b: 5,
+            c: 3
+          },
+          {
+            a: 6,
+            b: 6,
+            c: 3
+          }
+        ]
+      }
+    })
+
+    const filter: QueryFilter = {
+      or: [
+        {
+          a: {
+            eq: 1
+          },
+          or: [
+            {
+              c: {
+                eq: 6
+              }
+            }, {
+              b: {
+                eq: 5
+              }
+            }
+          ]
+        }
+      ]
+    }
```

Comment:
This reads `either "a" equal 1 or ("c" equal 6 or "b" equal 5)`, right? 

Looking at the sample 
```ts
 [
          {
            a: 1,
            b: 5,
            c: 8
          },
          {
            a: 1,
            b: 2,
            c: 10
          },
          {
            a: 1,
            b: 5,
            c: 3
          },
          {
            a: 6,
            b: 6,
            c: 3
          }
        ]
```

I expected the query to return `3` items but the assertion `expect(items).toHaveLength(2);` seem to indicate two.

Reply:
This reads `a = 1 and (c equal 6 or b equal 5)`. Notice that there is only a single item in the first `or` array.

Reply:
> Notice that there is only a single item in the first or array.

I right. I missed this part, good catch! Thanks.